diff --git a/drivers/net/usb/lan78xx.c b/drivers/net/usb/lan78xx.c
index caf2aff13..62ea9222b 100644
--- a/drivers/net/usb/lan78xx.c
+++ b/drivers/net/usb/lan78xx.c
@@ -2781,6 +2781,8 @@ static int lan78xx_reset(struct lan78xx_net *dev)
 		return ret;
 
 	buf |= HW_CFG_MEF_;
+	buf |= HW_CFG_CLK125_EN_;
+	buf |= HW_CFG_REFCLK25_EN_;
 	ret = lan78xx_write_reg(dev, HW_CFG, buf);
 	if (ret < 0)
 		return ret;
